src/baggage/remote_baggage_restriction_manager.js

```diff
@@ -0,0 +1,161 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Metrics from '../metrics/metrics.js';
+import NullLogger from '../logger.js';
+import NoopMetricFactory from '../metrics/noop/metric_factory';
+import * as constants from './default_baggage_restriction_manager';
+import Utils from '../util.js';
+
+const DEFAULT_REFRESH_INTERVAL = 60000;
+const DEFAULT_BAGGAGE_HOST = '0.0.0.0';
+const DEFAULT_BAGGAGE_PORT = 5778;
+const DEFAULT_FAIL_CLOSED = false;
+
+export default class RemoteBaggageRestrictionManager {
+
+    _serviceName: string;
+    _logger: Logger;
+    _metrics: Metrics;
+
+    _refreshInterval: number;
+    _host: string;
+    _port: number;
+    _failClosed: boolean;
+    _restrictionsMap: { [key: string]: number };
+
+    _initialDelayTimeoutHandle: any;
+    _refreshIntervalHandle: any;
+    _onRestrictionsUpdate: ?Function;
+
+    _initialized: boolean;
+
+    /**
+     * Creates a BaggageRestrictionManager that fetches baggage restrictions remotely from jaeger-agent.
+     *
+     * @param {string} [serviceName] - name of the current service / application, same as given to Tracer
+     * @param {object} [options] - optional settings
+     * @param {object} [options.logger] - optional logger, see _flow/logger.js
+     * @param {object} [options.failClosed] - this determines the startup failure mode of RemoteBaggageRestrictionManager.
+     * If failClosed is true, RemoteBaggageRestrictionManager will not allow any baggage to be written until baggage restrictions
+     * have been retrieved from agent. If failClosed is false (ie FailOpen), RemoteBaggageRestrictionManager will allow any baggage
+     * to be written until baggage restrictions have been retrieved from agent.
+     * @param {object} [options.metrics] - instance of Metrics object
+     * @param {number} [options.refreshInterval] - interval in milliseconds before baggage restrictions refreshes (0 to not refresh)
+     * @param {string} [options.host] - host for jaeger-agent, defaults to 'localhost'
+     * @param {number} [options.port] - port for jaeger-agent for BaggageRestrictionManager endpoint
+     * @param {function} [options.onRestrictionsUpdate]
+     */
+    constructor(serviceName: string, options: any = {}) {
+        this._serviceName = serviceName;
+        this._logger = options.logger || new NullLogger();
+        this._metrics = options.metrics || new Metrics(new NoopMetricFactory());
+        this._refreshInterval = options.refreshInterval || DEFAULT_REFRESH_INTERVAL;
+        this._host = options.host || DEFAULT_BAGGAGE_HOST;
+        this._port = options.port || DEFAULT_BAGGAGE_PORT;
+        this._failClosed = options.failClosed || DEFAULT_FAIL_CLOSED;
+        this._restrictionsMap = {};
+        this._initialized = false;
+        this._onRestrictionsUpdate = options.onRestrictionsUpdate;
+
+        this._refreshBaggageRestrictions();
```

Comment:
the baggage restrictions cache should be initialized (or at least attempted) before this constructor completes. This way when the tracer is initialized, users can start adding baggage right away rather than having to wait for at most 60 seconds for the initial delay.